collect_app/src/main/java/org/odk/collect/android/backgroundwork/FormUpdateAndInstanceSubmitScheduler.java

```diff
@@ -68,19 +68,31 @@ public void cancelUpdates(String projectId) {
     }
 
     @Override
-    public void scheduleSubmit(String projectId) {
-        Scheduler.NetworkType networkType = null;
+    public void scheduleAutoSend(String projectId) {
+        Scheduler.NetworkType networkConstraint;
         Settings settings = settingsProvider.getUnprotectedSettings(projectId);
         AutoSend autoSendSetting = StringIdEnumUtils.getAutoSend(settings, application);
         if (autoSendSetting == AutoSend.WIFI_ONLY) {
-            networkType = Scheduler.NetworkType.WIFI;
+            networkConstraint = Scheduler.NetworkType.WIFI;
         } else if (autoSendSetting == AutoSend.CELLULAR_ONLY) {
-            networkType = Scheduler.NetworkType.CELLULAR;
+            networkConstraint = Scheduler.NetworkType.CELLULAR;
+        } else if (autoSendSetting == AutoSend.WIFI_AND_CELLULAR) {
+            networkConstraint = null;
+        } else {
+            return;
         }
 
         HashMap<String, String> inputData = new HashMap<>();
         inputData.put(TaskData.DATA_PROJECT_ID, projectId);
-        scheduler.networkDeferred(getAutoSendTag(projectId), new AutoSendTaskSpec(), inputData, networkType);
+        scheduler.networkDeferred(getAutoSendTag(projectId), new SendFormsTaskSpec(), inputData, networkConstraint);
```

Comment:
Potentially, I'd need to experiment with `TaskSpecWorker` using a Kotlin object instead of a class (the way the reflection would work would be different if there's no constructor). Let's look at that when we're next playing with async stuff?

Reply:
Ok.